src/order/succ_pred/basic.lean

```diff
@@ -975,23 +975,24 @@ end linear_order
 /-- A strictly monotonous function from `no_max_order` to `is_succ_archimedean` is unbounded. -/
 lemma unbounded_of_strict_mono
   [preorder α] [preorder β]
-  [no_max_order α] [inhabited α]
+  [no_max_order α] [nonempty α]
   [succ_order β] [is_succ_archimedean β]
   (f : α → β) (Hmono : strict_mono f) :
-  set.unbounded (≤) (set.range f) :=
+  ¬ bdd_above (set.range f) :=
 begin
-  by_contra',
-  have bounded : set.bounded (≤) (set.range f) := by rwa set.not_unbounded_iff at this,
+  assume bounded : bdd_above (set.range f),
   obtain ⟨m, Hm⟩ : ∃ m, ∀ v ∈ set.range f, v ≤ m := bounded,
   have Hm' : ∀ a, f a ≤ m := λ a, Hm (f a) (set.mem_range_self _),
-  let v₀ := f default,
+  apply nonempty.elim ‹nonempty α›,
+  assume (a₀ : α),
```

Comment:
Is this the preferred way of making use of a `nonempty` assumption? I had a hard time finding existing examples...

Reply:
You should use `classical.arbitrary`.

Reply:
Huh, even though currently it doesn't use anything classical?

Reply:
That's because the goal is a `Prop`, meaning that you can use `nonempty.elim`. But genuinely we do not care about proof computability so better use `classical.arbitrary` directly.